app/src/heartbeat/heartbeat_storage_desktop.h

```diff
@@ -61,7 +68,13 @@ class HeartbeatStorageDesktop {
       const LoggedHeartbeats& heartbeats_struct) const;
 
   // local variables for state
+#if FIREBASE_PLATFORM_WINDOWS
+  // Use a wide string on Windows, to support international characters in the
+  // path.
+  std::wstring filename_;
```

Comment:
Does code like this just work as-is with wstring?
std::ifstream file(filename_,

Reply:
Yep, on Windows you can initialize an fstream with a wstring path. (You can't do this on non-Windows systems.)